#1. Merge newly created Stack class

rrajasek95 opened the pull request:
The current format for designing data structures I'm following consists of the following:
- A set of well defined primitives (Node objects for LinkedLists and Binary Trees)
- A Data Structure class which performs operations using the primitives as the units e.g. push(val), pop() for a Stack class using the ListNode primitive.

I have updated `stack.py` to reflect this new convention. Hopefully it's a lot cleaner than creating functions accepting an object as a parameter.

rrajasek95 committed:
x
Create primitives.py
Modify stack.py

mani1996 commented:
Remove one of the **init** definitions in the ListNode class. Namespace holds only the latest definition anyway. Also, use the del command and delete the object during pop operation. 

Yeah. This format looks better. I will start modifying the others based on this.

rrajasek95 committed:
Removed redundant init definition

rrajasek95 committed:
Remove underscore from member variable names
Remove getter/setter methods from the classes
Convert getter/setter method calls to member variable access
